runtime/src/main/kotlin/kotlin/Any.kt

```diff
@@ -41,10 +42,9 @@ public open class Any {
      */
     public open fun toString(): String {
         val kClass = this::class
-        val className = kClass.qualifiedName ?: kClass.simpleName ?: "<object>"
         // TODO: consider using [identityHashCode].
-        val unsignedHashCode = this.hashCode().toLong() and 0xffffffffL
-        val hashCodeStr = unsignedHashCode.toString(16)
-        return "$className@$hashCodeStr"
+        val identity = this.hashCode()
+        // Use debug description as the default string representation.
+        return debugDescription(kClass, identity)
```

Comment:
`Any.toString()` is not exactly the same (abstraction) as debug description.
I would prefer this patch to be reverted, code duplication is intentional here.

Reply:
Okay, I just thought that the default `toString()` implementation is exactly the debug description of an object.

Reply:
Just a coincidence :)